### PR TITLE
Alter the student List - Making Header stick to the top

### LINKED ISSUE #1469: In mobile, making sure key information is accessible as the user scrolls
In the Course Editors Page, the header of the page should stick to the top while user is scrolling down -- to help them to understand the information being displayed.
**Current Version**
Before scrolling
scroll_stick2
After scrolling
52stick_row_tothetop_editorssection
**Improvement**
This feature would make it easy to understand the information displayed in the table.
**Development**
Work involves fixing the table header at the top when user scrolls it out of view.

### PR DESCRIPTION
**Pull request by sejalkhatri**
The header of the page sticks to the top when the user scrolls down -- to help them understand the information being displayed. Fixes #1469

- Added a StickyHeader bool PropType to List Component which can be set to true if this functionality is required.

[
stickyheader
](url)

### CODE REVIEW

**Comment by ragesoss** [comment]
The positioning can shift when it switches into sticky mode.

Before sticking:
normal

After sticking:
offset

**Comment by ragesoss** [comment]
Is it possible to do this with the `Affix` component, as we do with CourseNavbar in the `return` call in `course.jsx`?

**Comment by sejalkhatri** [comment]
I don't think it's possible because Affix component is wrapped around a complete section which has to be fixed. 
In this particular case we don't want the entire List to be fixed but just the Table header.

File: app/assets/javascripts/components/common/list.jsx [hunk 1]
```diff
@@ -19,18 +19,19 @@ const List = createReactClass({
...
+    if (persistAreaElements.length !== 0) {
+      for (let i = 0; i < persistAreaElements.length; i++) {
+        const persistHeaderRow = persistAreaElements[i].getElementsByClassName('persist-header');
+        const clonedHeaderRow = persistHeaderRow[0].cloneNode(true);
+        const style = window.getComputedStyle(persistHeaderRow[0]);
+        const width = style.getPropertyValue('width');
+        clonedHeaderRow.style.width = width;
+        clonedHeaderRow.classList.add("floatingHeader");
+        persistAreaElements[i].childNodes[0].insertBefore(clonedHeaderRow, persistHeaderRow[0]);
+      }
```
Comment (sejalkhatri) [inline app/assets/javascripts/components/common/list.jsx]: @ragesoss Is this what you were suggesting? I have refactored some of the code but I don't see a point why not to use JQuery!!

File: app/assets/javascripts/components/revisions/diff_viewer.jsx [hunk 2]
```diff
@@ -54,6 +54,7 @@ const DiffViewer = createReactClass({
 
   showDiff() {
     this.setState({ showDiff: true });
+    $('.floatingHeader').css({ opacity: 0 });
```
Comment (ragesoss) [inline app/assets/javascripts/components/revisions/diff_viewer.jsx]: Eek! Any way to avoid putting workarounds in unrelated components?

**Comment by sejalkhatri** [comment]
Changing z-index does not work because both the divs are in different stacking context. So, I have tried to fix the problem by changing the styling. Let me know if you can think of another workaround.

**Comment by ragesoss** [comment]
That stacking context thing is no end of trouble! This is okay with me.

**Pull request merged**